File: lib/larb.rb
# frozen_string_literal: true

begin
  require "larb/larb"
rescue LoadError
  # Fallback to the Ruby implementation when the extension isn't built yet.
end

require_relative "larb/vec2" unless defined?(Larb::Vec2)
require_relative "larb/vec3" unless defined?(Larb::Vec3)
require_relative "larb/vec4" unless defined?(Larb::Vec4)
require_relative "larb/quat" unless defined?(Larb::Quat)
require_relative "larb/quat2" unless defined?(Larb::Quat2)
require_relative "larb/mat2" unless defined?(Larb::Mat2)
require_relative "larb/mat2d" unless defined?(Larb::Mat2d)
require_relative "larb/mat3" unless defined?(Larb::Mat3)
require_relative "larb/mat4" unless defined?(Larb::Mat4)
require_relative "larb/color" unless defined?(Larb::Color)

module Larb
  VERSION = "0.1.0"
end

File: ext/larb/larb.c
#include "larb.h"
#include "vec2.h"
#include "vec3.h"
#include "vec4.h"
#include "mat2.h"
#include "mat3.h"
#include "mat4.h"
#include "quat.h"
#include "mat2d.h"

VALUE mLarb = Qnil;

void Init_larb(void) {
  mLarb = rb_define_module("Larb");
  Init_vec2(mLarb);
  Init_vec3(mLarb);
  Init_vec4(mLarb);
  Init_mat2(mLarb);
  Init_mat3(mLarb);
  Init_quat(mLarb);
  Init_mat4(mLarb);
  Init_mat2d(mLarb);
}
